File: 01_insertion_sort/ex01_insertion_sort_basico.c
/*
Exercicio 01
Tema: 01_insertion_sort
*/

void ordenar_crescente(int * arr, int tamanho){
    for(int i = 0; i < tamanho; i++){
        int temp = arr[i];
        int j = i - 1;

        while(j >= 0 && arr[j] > temp){
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;    
    }
}

#include <stdio.h>

int main() {

    int arr[] = {5, 2, 8, 1, 9};
    int tamanho = sizeof(arr) / sizeof(arr[0]);

    ordenar_crescente(arr, tamanho);

    for(int i = 0; i < tamanho; i++){
        printf("%d ", arr[i]);
    }

    return 0;
}

File: 01_insertion_sort/ex01_insertion_sort_basico.py
"""
Exercicio 01
Tema: 01_insertion_sort
"""

def ordenar_crescente(arr, tamanho):
    for i in range(1, tamanho):
        temp = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > temp:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = temp

arr = [5, 2, 8, 1, 9]

ordenar_crescente(arr, len(arr))

print(arr)
